rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1828,6 +1828,535 @@ func Qualify_Shaping: <"The qualification of a shaping event from the fact that
+	inputs:
+		tradeState TradeState (1..1) <"The original trade to be for substitution of collateral.">
+		effectiveDate AdjustableOrRelativeDate (1..1) <"The date to close and open a new trade with new collateral.">
+		newCollateralPortfolio CollateralPortfolio(1..1) <"New collateral portfolio to subtitute for the original collateral.">
+		priceQuantity PriceQuantity (1..*) <"The price and quantity of the substituted product.">
+	output: 
+		instruction PrimitiveInstruction (1..1)
+
+	condition ContractualProduct: <"Only a security finance contractual product can substitute collateral.">
+		tradeState -> trade -> tradableProduct -> product -> contractualProduct exists
+
+        set instruction: <"Sets the second part of the split to be a new contract with a new asset payout.">
+                PrimitiveInstruction {
+                    quantityChange: QuantityChangeInstruction {
+                            change: priceQuantity,
+                            direction: QuantityChangeDirectionEnum -> Replace,
+                            lotIdentifier: empty
+                        },
+                    termsChange: Create_SubstitutionInstruction(
+                                tradeState -> trade -> tradableProduct -> product -> contractualProduct,
+                               effectiveDate,
+                               newCollateralPortfolio
+                            ),
+                    ...
+                }
+
+func Create_SubstitutionInstruction:<"Creates the terms change instruction that updates the payout with the new subsitution payout.">
+	inputs:
+		contractualProduct ContractualProduct (1..1) <"The orginal contractual product to be used as the basis of the new trade.">
+		effectiveDate AdjustableOrRelativeDate (1..1) <"The date to close and open a new position.">
+		newCollateralPortfolio CollateralPortfolio(1..1) <"New collateral portfolio to subtitute for the original collateral.">
+	output:
+		termsChangeInstruction TermsChangeInstruction (1..1)
+
+	set termsChangeInstruction -> product -> contractualProduct:
+		contractualProduct
+	set termsChangeInstruction -> product -> contractualProduct -> economicTerms -> effectiveDate:
+		effectiveDate
+	set termsChangeInstruction -> product -> contractualProduct -> economicTerms -> collateral -> collateralPortfolio:
+		newCollateralPortfolio
+
+
+func Qualify_Substitution: <"Qualification of a collateral substitution event.">
+	[qualification BusinessEvent]
+	inputs:
+	businessEvent BusinessEvent (1..1)
+	output: is_event boolean (1..1)
+
+	alias beforeEconomicterms: businessEvent -> instruction only-element -> before -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms
+	alias openEconomicTerms: FilterOpenTradeStates( businessEvent -> after ) only-element -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms
+	alias closedTradeState: FilterClosedTradeStates( businessEvent -> after )
```

Comment:
Don't think we need this (of the check under `is_event` for it) as the termsChange does not close the trade - unless I am missing something?